From 20e6326555973801824779293d1e4d931b57c1b5 Mon Sep 17 00:00:00 2001
From: Jamin Lin <jamin_lin@aspeedtech.com>
Date: Wed, 16 Nov 2022 13:28:53 +0800
Subject: [PATCH] smbus: add to check root device map

If users issue “TriggerDeviceDiscovery” to scan the device and got the empty i2c-mux device, all device will be removed including root device.
However, AST2600 EVB did not have i2c-mux device. Besides, we used i2c bus 4 to connect AST1060 on DCSCM board and
AST1060 slave device will be added in the root device map when MCTP daemon start.
Add to check both “rootDeviceMap” and “registerDeviceMap” are empty, then remove all device.

Signed-off-by: Jamin Lin <jamin_lin@aspeedtech.com>
---
 src/SMBusBinding.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/SMBusBinding.cpp b/src/SMBusBinding.cpp
index dd61f15..a43827f 100644
--- a/src/SMBusBinding.cpp
+++ b/src/SMBusBinding.cpp
@@ -965,7 +965,8 @@ void SMBusBinding::initEndpointDiscovery(boost::asio::yield_context& yield)
         }
     }
 
-    if (registerDeviceMap.empty())
+    // Add to check root device
+    if (registerDeviceMap.empty() && rootDeviceMap.empty())
     {
         phosphor::logging::log<phosphor::logging::level::DEBUG>(
             "No device found");
-- 
2.25.1

